app/nexus/internal/state/backend/sqlite/persist/policy.go

```diff
@@ -61,6 +64,22 @@ func (s *DataStore) DeletePolicy(ctx context.Context, id string) error {
 	return nil
 }
 
+func GenerateCustomNonce(s *DataStore) ([]byte, error) {
+	nonce := make([]byte, s.Cipher.NonceSize())
+	if _, err := io.ReadFull(rand.Reader, nonce); err != nil {
+		return nil, err
+	}
+	return nonce, nil
+}
+
+func EncryptWithCustomNonce(s *DataStore, nonce []byte, data []byte) ([]byte, error) {
+	if len(nonce) != s.Cipher.NonceSize() {
+		return nil, fmt.Errorf("invalid nonce size: got %d, want %d", len(nonce), s.Cipher.NonceSize())
+	}
+	ciphertext := s.Cipher.Seal(nil, nonce, data, nil)
+	return ciphertext, nil
+}
+
```

Comment:
Why are these methods public? 
we only used them inside the policy package;
suggest making them private.